src/main/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProvider.java

```diff
@@ -44,6 +44,7 @@ public class PerformanceAnalyzerResourceProvider extends BaseRestHandler {
             LogManager.getLogger(PerformanceAnalyzerResourceProvider.class);
 
     private static final int HTTP_CLIENT_CONNECTION_TIMEOUT_MILLIS = 200;
+    private static final int HTTP_READ_CONNECTON_TIMEOUT_MILLIS = 10000;
```

Comment:
How did we arrive at 10 seconds. Does this introduce a problem/risk of any sorts?

Reply:
Typically in such queries, node needs to collect all metric info from other nodes and send back response. And usually in ideal case when nodes are healthy, this api returns well within a second. 
Came to 10 sec to keep it long enough to wait for response and short enough to timeout to avoid a situation where a lot of threads are being stuck at a particular point. 
Though I am open to your thoughts.

Reply:
I think that's a reasonable guesstimate for the assumption. LGTM +1